### PR TITLE
Lowercase username when managing secrets

### LINKED ISSUE #502: Uppercase username is invalid as part of sealed secret name
Account name with uppercase character are not valid to use as a sealed secret name and it cause issue when `import-secret` try to sealed a secret.

## Expected Behaviour

OFC should handle upper case user name. OFC lower the user name the time of importing and reading secret. 

## Current Behaviour

I got error when I created a secret with the name *qolzam-slack-token* :
> http status code 400, error: secrets *Qolzam-slack-token* not found

Then I created a secret with the name *Qolzam-slack-token* and redeploy but I got error again:

>[failed to parse secrets, error : import-secrets returned unexpected status: exit status 255
hash for HMAC validated successfully for Qolzam
couldn’t create SealedSecret (Qolzam-slack-token) - error: SealedSecret.bitnami.com “Qolzam-slack-token” is invalid: metadata.name: Invalid value: “Qolzam-slack-token”: a DNS-1123 subdomain must consist of lower case alphanumeric characters, ‘-’ or ‘.’, and must start and end with an alphanumeric character (e.g. ‘example.com’, regex used for validation is ‘a-z0-9?(.a-z0-9?)*’)
## Possible Solution

To work around creating an Github org with lowercase character solved the issue.

But possible solution could be lower the `Owner` name in `buildshiprun` and `import-secrets` handlers.
When it is adding owner name to secret from function YAML in [buildshiprun handler](https://github.com/openfaas/openfaas-cloud/blob/master/buildshiprun/handler.go#L461).

When it’s reading from secret.yml file to import the secrets in [import-secrets handler](https://github.com/openfaas/openfaas-cloud/blob/master/import-secrets/handler.go#L70)

## Steps to Reproduce (for bugs)

1. Have an Github account/org with upper case name.
2. [Use faas-cli](https://docs.openfaas.com/openfaas-cloud/secrets/) to create a sealed secret and add secret name to *stack.yml*.
3. Commit and push to Github.
4. See the log status in Github.

## Context

 
I want to create a secret for my function.

## Your Environment

* FaaS-CLI version ( Full output from: `faas-cli version` ): 0.8.13

* Docker version `docker version` (e.g. Docker 17.0.05 ): 19.03.1

* Are you using Docker Swarm or Kubernetes (FaaS-netes)? Kubernetes

* Operating System and version (e.g. Linux, Windows, MacOS): MacOS

* Link to your project or a code example to reproduce issue:

* Please also follow the [troubleshooting guide](https://github.com/openfaas/faas/blob/master/guide/troubleshooting.md) and paste in any other diagnostic information you have:

### PR DESCRIPTION
**Pull request by jonatasbaldin**
Signed-off-by: Jonatas Baldin <jonatas.baldin@gmail.com>

## Description
Lower case the username when managing secrets. Fixes #502.

Will still add documentation about creating the sealed secret just in lowercase, even if the GitHub username is uppercase.

## How Has This Been Tested?
- Using a GitHub user account with an uppercase character
- Create a sealed secret in lowercase
- Push the project and it builds correctly
- Tests made [here](https://github.com/Uppercase-username/ofc-test/runs/266808847)

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [x] read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [ ] added unit tests

### CODE REVIEW

File: buildshiprun/handler.go [hunk 1]
```diff
@@ -458,7 +458,7 @@ func getEventFromEnv() (*sdk.Event, error) {
 	info.Secrets = secretVars
 
 	for i := 0; i < len(info.Secrets); i++ {
-		info.Secrets[i] = info.Owner + "-" + info.Secrets[i]
+		info.Secrets[i] = strings.ToLower(info.Owner) + "-" + info.Secrets[i]
```
Comment (alexellis) [inline buildshiprun/handler.go]: Let's set `owner :=` outside the loop

File: import-secrets/handler.go [hunk 2]
```diff
@@ -67,12 +67,12 @@ func Handle(req []byte) string {
 		return fmt.Sprintf("invalid owner name %s", event.owner)
 	}
 
-	if strings.HasPrefix(userSecret.Metadata.Name, event.owner) == false {
+	name := strings.ToLower(fmt.Sprintf("%s", userSecret.Metadata.Name))
```
Comment (alexellis) [inline import-secrets/handler.go]: Why do we need sprint here?

File: buildshiprun/handler.go (hunk 1 above)
Comment (jonatasbaldin) [reply buildshiprun/handler.go]: Fixed.

File: import-secrets/handler.go (hunk 2 above)
Comment (jonatasbaldin) [reply import-secrets/handler.go]: 🤷‍♀️ it was like this before hehe

but removed it 💃

File: import-secrets/handler.go [hunk 3]
```diff
@@ -67,12 +67,12 @@ func Handle(req []byte) string {
...
+
+	if strings.HasPrefix(name, strings.ToLower(event.owner)) == false {
```
Comment (alexellis) [inline import-secrets/handler.go]: I don't think we need to toLower both parts:

```
	ownerNormalized := strings.ToLower(event.owner)

	if strings.HasPrefix(userSecret.Metadata.Name, ownerNormalized) == false {
		return fmt.Errorf("unable to bind a secret which does not start with owner name: %s", ownerNormalized).Error()
	}
```

**Comment by jonatasbaldin** [comment]
Fixing the documentation here: https://github.com/openfaas/docs/pull/184

File: import-secrets/handler.go (hunk 3 above)
Comment (jonatasbaldin) [reply import-secrets/handler.go]: Fixed as well!

**Comment by alexellis** [comment]
LGTM, thank you

**Pull request merged**